backend/habragasolina/src/main/java/bo/edu/usfa/gasolina/habragasolina/Service/UserService.java

```diff
@@ -0,0 +1,76 @@
+package bo.edu.usfa.gasolina.habragasolina.Service;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import bo.edu.usfa.gasolina.habragasolina.Entities.User;
+import bo.edu.usfa.gasolina.habragasolina.Repository.GasStationRepository;
+import bo.edu.usfa.gasolina.habragasolina.Repository.UserRepository;
+
+@Service
+public class UserService {
+    
+    private final UserRepository userRepository;
+    
+    private final GasStationRepository gasStationRepository;
+
+    public UserService(UserRepository userRepository, GasStationRepository gasStationRepository) {
+        this.userRepository = userRepository;
+        this.gasStationRepository = gasStationRepository;
+    }
+    
+    public User saveUser(User user) {
+
+        //validations
+        if(user.getPassword().length() <= 6){
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Password must be longer or equal than 6 characters");
+        }
+        
+        if(userRepository.existsByUsername(user.getUsername())){
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "User is duplicated");
+        }
+
+        if(user.getUsername().length() <= 3){
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "UserName must be longer than 3 characters");
```

Comment:
Change the validation response to be Bad Request, not found in this endpoint would only be used for the Id Gas Station validation